torch/_higher_order_ops/cond.py

```diff
@@ -137,10 +138,12 @@ def false_fn(x: torch.Tensor):
     )
 
     if isinstance(pred, (bool, int, float)):
-        log.warning(
-            "Pred is a Python constant. When used with torch.cond, it executes only one of the branches."
-            " If you want torch.cond to perserve two branches, please make the predicate a boolean tensor or a SymBool."
-        )
+        if torch.compiler.is_compiling():
+            warnings.warn(
+                "Pred is a Python constant. When exporting torch.cond, it exports one of the branches."
```

Comment:
ditto here: let's make it clearer that we're just going to specialize on one of the branches.